File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/mappers/ListMapping.java
package com.themaker.fshmo.klassikaplus.data.mappers;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

public class ListMapping<TFrom, TTo> {

    @NonNull
    private final Mapping<TFrom, TTo> mapping;

    public ListMapping(@NonNull final Mapping<TFrom, TTo> mapping) {
        this.mapping = mapping;
    }

    @NonNull
    public List<TTo> map(@NonNull final List<TFrom> list) {
        final ArrayList<TTo> arrayList = new ArrayList<>(list.size());
        for (final TFrom from : list) {
            arrayList.add(mapping.map(from));
        }
        return arrayList;
    }
}

File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/mappers/ListMapping.kt
package com.themaker.fshmo.klassikaplus.data.mappers

import java.util.ArrayList

class ListMapping<TFrom, TTo>(private val mapping: Mapping<TFrom, TTo>) {

    fun map(list: List<TFrom>): List<TTo> {
        val arrayList = ArrayList<TTo>(list.size)
        for (from in list) {
            arrayList.add(mapping.map(from))
        }
        return arrayList
    }
}